cherab/openadas/rates/beam.pyx

```diff
@@ -19,10 +19,31 @@
 from cherab.core.utility.conversion import PhotonToJ
 
 cimport cython
-from cherab.core.math cimport Interpolate1DCubic, Interpolate2DCubic
+from libc.math cimport INFINITY
+from raysect.core.math.function.float cimport Interpolator1DArray, Interpolator2DArray, Constant1D, Constant2D
 
 # todo: clarify variables
 
+cdef class FunctionY2D(Function2D):
+    cdef Function1D _function
+
+    def __init__(self, Function1D function):
+        self._function = function
+
+    cdef double evaluate(self, double x, double y) except? -1e999:
+        return self._function(y)
+
+
+cdef class FunctionX2D(Function2D):
+    cdef Function1D _function
+
+    def __init__(self, Function1D function):
+        self._function = function
+
+    cdef double evaluate(self, double x, double y) except? -1e999:
+        return self._function(x)
+
+
```

Comment:
If you are to use these, you should at least call `self._function.evaluate(x)` to avoid going through the Python `__call__` layer.

But, I think you can already accomplish this without adding to the zoo of function classes using `cherab.core.math.mappers.IsoMapper2D` and `raysect.core.math.function.float.Arg2D`:

```python
f1d_interpolator = Interpolator1DArray(...)
f2d_x_only = IsoMapper2D(Arg2D('x'), f1d_interpolator)
f2d_y_only = IsoMapper2D(Arg2D('y'), f1d_interpolator)
```

Reply:
I didn’t think of this solution, thanks. Fixed now.